apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/BranchController.java

```diff
@@ -0,0 +1,139 @@
+import com.ctrip.framework.apollo.portal.component.PermissionValidator;
+import com.ctrip.framework.apollo.portal.component.config.PortalConfig;
+import com.ctrip.framework.apollo.portal.entity.bo.NamespaceBO;
+import com.ctrip.framework.apollo.portal.entity.model.NamespaceReleaseModel;
+import com.ctrip.framework.apollo.portal.listener.ConfigPublishEvent;
+import com.ctrip.framework.apollo.portal.service.NamespaceBranchService;
+import com.ctrip.framework.apollo.portal.service.ReleaseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+@RequestMapping("/openapi/v1/envs/{env}")
+public class BranchController {
+
+    @Autowired
+    private PermissionValidator permissionValidator;
+    @Autowired
+    private ReleaseService releaseService;
+    @Autowired
+    private NamespaceBranchService namespaceBranchService;
+    @Autowired
+    private ApplicationEventPublisher publisher;
+    @Autowired
+    private PortalConfig portalConfig;
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches", method = RequestMethod.GET)
+    public NamespaceBO findBranch(@PathVariable String appId,
+                                  @PathVariable String env,
+                                  @PathVariable String clusterName,
+                                  @PathVariable String namespaceName) {
+
+        return namespaceBranchService.findBranch(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName);
+    }
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/operator/{operator}/branches", method = RequestMethod.POST)
+    public NamespaceDTO createBranch(@PathVariable String appId,
+                                     @PathVariable String env,
+                                     @PathVariable String clusterName,
+                                     @PathVariable String namespaceName,
+                                     @PathVariable String operator,
+                                     HttpServletRequest request) {
+
+        return namespaceBranchService.createBranch(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName, operator);
+    }
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/operator/{operator}", method = RequestMethod.DELETE)
+    public void deleteBranch(@PathVariable String appId,
+                             @PathVariable String env,
+                             @PathVariable String clusterName,
+                             @PathVariable String namespaceName,
+                             @PathVariable String branchName,
+                             @PathVariable String operator) {
```

Comment:
normally, operator should be a RequestParam(query parameter) instead of a PathVariable